crates/sui-core/src/authority.rs

```diff
@@ -1616,18 +1612,17 @@ impl AuthorityState {
         Ok(())
     }
 
-    #[allow(dead_code)]
-    pub(crate) fn update_committee(&self, new_committee: Committee) -> SuiResult {
-        // TODO: It's likely safer to do the following operations atomically, in case this function
-        // gets called from different threads. It cannot happen today, but worth the caution.
+    pub fn reconfigure(&self, new_committee: Committee) -> SuiResult {
+        // TODO: We should move the committee into epoch db store, so that the operation below
+        // can become atomic.
         fp_ensure!(
             self.epoch() + 1 == new_committee.epoch,
             SuiError::from("Invalid new epoch to sign and update")
         );
 
         self.committee_store.insert_new_committee(&new_committee)?;
-        // TODO: Do we want to make it possible to subscribe to committee changes?
         self.committee.swap(Arc::new(new_committee));
+        self.db().reopen_epoch_db(self.epoch());
```

Comment:
Out of curiosity, do we drop at any point the previous epoch db?

Reply:
Good question. Not yet. But it should be trivial to do.

Reply:
Yep that's my guess as well. For Narwhal we do also want to drop the previous db, but I am not convinced whether this should be done in a sync way, or choose an async approach as data set can be  quite large and this can take some time. If operation is fast in any case then we can choose the simplest way